sklearn/decomposition/ksvd.py

```diff
@@ -0,0 +1,343 @@
+"""K-SVD dictionary learning algorithm.
+"""
+
```

Comment:
+1 for fusing implementations. Then we can see if we need an extra ksvd
function. I think it's not necessary if it's documented. It also helps
users to realize that KSVD is not fundamentally different from other
dictionary learning approaches.